### PR TITLE
[Workers] compat date clarification

### PR DESCRIPTION
**Pull request by tanushree-sharma**

### CODE REVIEW

File: content/workers/configuration/compatibility-dates.md [hunk 1]
```diff
@@ -33,21 +27,61 @@ However, even though you do not need to update the `compatibility_date` field, i
...
+
+## Setting compatibility flags
+
+In addition to setting a `compatibility_date`, you may also provide a list of `compatibility_flags`, which enable or disable specific changes.
+
+Most developers will not need to use `compatibility_flags`; instead, Cloudflare recommends only specifying `compatibility_date`. `compatibility_flags` can be useful if you want to help the Workers team test upcoming changes that are not yet enabled by default, or if you need to hold back a change that your code depends on but still want to apply other compatibility changes.
+
+#### Via Wrangler
+
+The compatability flags can be set in a Worker's [`wrangler.toml`](/workers/wrangler/configuration/) file.
```
Comment (irvinebroque) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
Compatability flags can be set in a Worker's [`wrangler.toml`](/workers/wrangler/configuration/) file.
```

File: content/workers/configuration/compatibility-dates.md [hunk 2]
```diff
@@ -33,21 +27,61 @@ However, even though you do not need to update the `compatibility_date` field, i
...
 compatibility_flags = [ "formdata_parser_supports_files" ]
 ```
 
-This example enabled the specific flag `formdata_parser_supports_files`, which is described below. As of the specified date, `2021-09-14`, this particular flag was not yet enabled by default, but specifying it in this way enables it anyway. `compatibility_flags` can also be used to disable changes that became the default in the past.
+#### Via Workers Dashboard
 
-Most developers will not need to use `compatibility_flags`; instead, Cloudflare recommends only specifying `compatibility_date`. `compatibility_flags` can be useful if you want to help the Workers team test upcoming changes that are not yet enabled by default, or if you need to hold back a change that your code depends on but still want to apply other compatibility changes.
+Compatability flags cannot be set or updated on the Workers dashboard at this time. 
+
+#### Via API
```
Comment (irvinebroque) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
#### Via the Cloudflare API
```

File: content/workers/configuration/compatibility-dates.md [hunk 3]
```diff
@@ -33,21 +27,61 @@ However, even though you do not need to update the `compatibility_date` field, i
...
+
+Compatability flags can be set when uploading a Worker using the [Workers Script API](/api/operations/worker-script-upload-worker-module) or [Workers Versions API](/api/operations/worker-versions-upload-version#request-body) in the request body's metadata field.
```
Comment (irvinebroque) [inline content/workers/configuration/compatibility-dates.md]: It's not clear from the API docs what the valid metadata fields are, ex:

https://developers.cloudflare.com/api/operations/worker-script-upload-worker-module

File: content/workers/configuration/compatibility-dates.md [hunk 4]
```diff
@@ -14,15 +14,9 @@ meta:
 
 Cloudflare regularly updates the Workers runtime. These updates apply to all Workers globally and should never cause a Worker that is already deployed to stop functioning. Sometimes, though, some changes may be backwards-incompatible. In particular, there might be bugs in the runtime API that existing Workers may inadvertently depend upon. Cloudflare implements bug fixes that new Workers can opt into while existing Workers will continue to see the buggy behavior to prevent breaking deployed Workers.
 
-Compatibility dates (and flags) are how you, as a developer, opt into these changes. By specifying a `compatibility_date` in your `wrangler.toml` file, that Worker enables all changes that were made before the given date.
+The compatibility date (and flags) are how you, as a developer, opt into these changes. Specifying a `compatibility_date` enables all changes that were made before the given date to the Worker.
```
Comment (GregBrimble) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
The compatibility date (and flags) are how you, as a developer, opt into these runtime changes. Compatibility flags will often have a date in which they are enabled by default, and so, by specifying a `compatibility_date` for your Worker, you can quickly enable all of these various compatibility flags up to, and including, that date.
```

File: content/workers/configuration/compatibility-dates.md [hunk 5]
```diff
@@ -33,21 +27,61 @@ However, even though you do not need to update the `compatibility_date` field, i
 1.  Sometimes, new features can only be made available to Workers that have a current `compatibility_date`. To access the latest features, you need to stay up-to-date.
 2.  Generally, other than this page, the Workers documentation may only describe the current `compatibility_date`, omitting information about historical behavior. If your Worker uses an old `compatibility_date`, you will need to continuously refer to this page in order to check if any of the APIs you are using have changed.
 
-## Compatibility flags
+#### Via Wrangler
+
+The compatability date can be set in a Worker's [`wrangler.toml`](/workers/wrangler/configuration/) file. The first deployment of your Worker, [`npx wrangler deploy`](/workers/wrangler/commands/#deploy), will automatically set the compatability date to the current date.
```
Comment (GregBrimble) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
The compatibility date can be set in a Worker's [`wrangler.toml`](/workers/wrangler/configuration/) file.
```

File: content/workers/configuration/compatibility-dates.md [hunk 6]
```diff
@@ -14,15 +14,9 @@ meta:
...
+The compatibility date (and flags) are how you, as a developer, opt into these changes. Specifying a `compatibility_date` enables all changes that were made before the given date to the Worker.
 
-Compatibility dates can only be set via your `wrangler.toml` file and by running [`npx wrangler deploy`](/workers/wrangler/commands/#deploy).
-
-```toml
-# (in wrangler.toml)
-# Opt into backwards-incompatible changes through April 5, 2022.
-compatibility_date = "2022-04-05"
-```
+## Setting compatability date
```
Comment (GregBrimble) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
## Setting compatibility date
```

File: content/workers/configuration/compatibility-dates.md [hunk 7]
```diff
@@ -33,21 +27,61 @@ However, even though you do not need to update the `compatibility_date` field, i
 1.  Sometimes, new features can only be made available to Workers that have a current `compatibility_date`. To access the latest features, you need to stay up-to-date.
 2.  Generally, other than this page, the Workers documentation may only describe the current `compatibility_date`, omitting information about historical behavior. If your Worker uses an old `compatibility_date`, you will need to continuously refer to this page in order to check if any of the APIs you are using have changed.
 
-## Compatibility flags
+#### Via Wrangler
+
+The compatability date can be set in a Worker's [`wrangler.toml`](/workers/wrangler/configuration/) file. The first deployment of your Worker, [`npx wrangler deploy`](/workers/wrangler/commands/#deploy), will automatically set the compatability date to the current date. 
+
+```sh
```
Comment (GregBrimble) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
```toml
```

File: content/workers/configuration/compatibility-dates.md [hunk 8]
```diff
@@ -33,21 +27,61 @@ However, even though you do not need to update the `compatibility_date` field, i
...
+---
+filename: wrangler.toml
+---
+# Opt into backwards-incompatible changes through April 5, 2022.
+compatibility_date = "2022-04-05"
+```
+
+#### Via Workers Dashboard
+
+When a Worker is created through the Workers Dashboard, the compatibility date is automatically set to the current date.
```
Comment (GregBrimble) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
#### Via the Cloudflare Dashboard

When a Worker is created through the Cloudflare Dashboard, the compatibility date is automatically set to the current date.
```

File: content/workers/configuration/compatibility-dates.md [hunk 9]
```diff
@@ -33,21 +27,61 @@ However, even though you do not need to update the `compatibility_date` field, i
...
+
+The compatability date cannot be updated on the Workers dashboard at this time.
```
Comment (GregBrimble) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
The compatibility date cannot be updated on the Cloudflare Dashboard at this time. 
```

File: content/workers/configuration/compatibility-dates.md [hunk 10]
```diff
@@ -33,21 +27,61 @@ However, even though you do not need to update the `compatibility_date` field, i
...
+When a Worker is created through the Workers Dashboard, the compatibility date is automatically set to the current date.
+
+The compatability date cannot be updated on the Workers dashboard at this time. 
 
-In addition to setting a `compatibility_date` in your `wrangler.toml` file, you may also provide a list of `compatibility_flags`, which enable or disable specific changes.
+#### Via API
+
+The compatability date can be set when uploading a Worker using the [Workers Script API](/api/operations/worker-script-upload-worker-module) or [Workers Versions API](/api/operations/worker-versions-upload-version#request-body) in the request body's metadata field. 
+
+If a compatability date is not specified on upload via API, it defaults to the oldest compatibility date, before any flags took effect (2021-11-02). It is highly reccomended to set the compatability date to the current date when uploading via API.
```
Comment (GregBrimble) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
The compatibility date can be set when uploading a Worker using the [Workers Script API](/api/operations/worker-script-upload-worker-module) or [Workers Versions API](/api/operations/worker-versions-upload-version#request-body) in the request body's `metadata` field. 

If a compatibility date is not specified on upload via the API, it defaults to the oldest compatibility date, before any flags took effect (2021-11-02). When creating new Workers, it is highly recommended to set the compatibility date to the current date when uploading via the API.
```

File: content/workers/configuration/compatibility-dates.md [hunk 11]
```diff
@@ -33,21 +27,61 @@ However, even though you do not need to update the `compatibility_date` field, i
...
+
+The compatability date can be set when uploading a Worker using the [Workers Script API](/api/operations/worker-script-upload-worker-module) or [Workers Versions API](/api/operations/worker-versions-upload-version#request-body) in the request body's metadata field. 
+
+If a compatability date is not specified on upload via API, it defaults to the oldest compatibility date, before any flags took effect (2021-11-02). It is highly reccomended to set the compatability date to the current date when uploading via API. 
+
+## Setting compatibility flags
+
+In addition to setting a `compatibility_date`, you may also provide a list of `compatibility_flags`, which enable or disable specific changes.
+
+Most developers will not need to use `compatibility_flags`; instead, Cloudflare recommends only specifying `compatibility_date`. `compatibility_flags` can be useful if you want to help the Workers team test upcoming changes that are not yet enabled by default, or if you need to hold back a change that your code depends on but still want to apply other compatibility changes.
```
Comment (GregBrimble) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
Most developers will not need to use `compatibility_flags`. `compatibility_flags` can be useful if you want to help the Workers team test upcoming changes that are not yet enabled by default, or if you need to hold back a change that your code depends on but still want to apply other compatibility changes.
```

File: content/workers/configuration/compatibility-dates.md (hunk 1 above)
Comment (GregBrimble) [reply content/workers/configuration/compatibility-dates.md]:
```suggestion
Compatibility flags can be set in a Worker's [`wrangler.toml`](/workers/wrangler/configuration/) file.
```

File: content/workers/configuration/compatibility-dates.md [hunk 12]
```diff
@@ -33,21 +27,61 @@ However, even though you do not need to update the `compatibility_date` field, i
...
+
+#### Via Wrangler
+
+The compatability flags can be set in a Worker's [`wrangler.toml`](/workers/wrangler/configuration/) file.
+
+This example enables the specific flag `formdata_parser_supports_files`, which is described [below](/workers/configuration/compatibility-dates/#formdata-parsing-supports-file). As of the specified date, `2021-09-14`, this particular flag was not yet enabled by default, but specifying it in this way enables it anyway. `compatibility_flags` can also be used to disable changes that became the default in the past.
```
Comment (GregBrimble) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
This example enables the specific flag `formdata_parser_supports_files`, which is described [below](/workers/configuration/compatibility-dates/#formdata-parsing-supports-file). As of the specified date, `2021-09-14`, this particular flag was not yet enabled by default, but, by specifying it in `compatibility_flags`, we can enable it anyway. `compatibility_flags` can also be used to disable changes that became the default in the past.
```

File: content/workers/configuration/compatibility-dates.md [hunk 13]
```diff
@@ -33,21 +27,61 @@ However, even though you do not need to update the `compatibility_date` field, i
...
+filename: wrangler.toml
+---
 # Opt into backwards-incompatible changes through September 14, 2021.
 compatibility_date = "2021-09-14"
 # Also opt into an upcoming fix to the FormData API.
 compatibility_flags = [ "formdata_parser_supports_files" ]
 ```
 
-This example enabled the specific flag `formdata_parser_supports_files`, which is described below. As of the specified date, `2021-09-14`, this particular flag was not yet enabled by default, but specifying it in this way enables it anyway. `compatibility_flags` can also be used to disable changes that became the default in the past.
+#### Via Workers Dashboard
```
Comment (GregBrimble) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
#### Via the Cloudflare Dashboard
```

File: content/workers/configuration/compatibility-dates.md [hunk 14]
```diff
@@ -33,21 +27,61 @@ However, even though you do not need to update the `compatibility_date` field, i
...
 
-Most developers will not need to use `compatibility_flags`; instead, Cloudflare recommends only specifying `compatibility_date`. `compatibility_flags` can be useful if you want to help the Workers team test upcoming changes that are not yet enabled by default, or if you need to hold back a change that your code depends on but still want to apply other compatibility changes.
+Compatability flags cannot be set or updated on the Workers dashboard at this time.
```
Comment (GregBrimble) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
Compatibility flags cannot be set or updated on the Cloudflare Dashboard at this time. 
```

File: content/workers/configuration/compatibility-dates.md [hunk 15]
```diff
@@ -33,21 +27,61 @@ However, even though you do not need to update the `compatibility_date` field, i
...
+```
+
+#### Via Workers Dashboard
+
+When a Worker is created through the Workers Dashboard, the compatibility date is automatically set to the current date.
+
+The compatability date cannot be updated on the Workers dashboard at this time. 
 
-In addition to setting a `compatibility_date` in your `wrangler.toml` file, you may also provide a list of `compatibility_flags`, which enable or disable specific changes.
+#### Via API
```
Comment (GregBrimble) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
#### Via the Cloudflare API
```

File: content/workers/configuration/compatibility-dates.md (hunk 3 above)
Comment (GregBrimble) [inline content/workers/configuration/compatibility-dates.md]:
```suggestion
Compatibility flags can be set when uploading a Worker using the [Workers Script API](/api/operations/worker-script-upload-worker-module) or [Workers Versions API](/api/operations/worker-versions-upload-version#request-body) in the request body's `metadata` field. 
```

**Review by GregBrimble — APPROVED** [review]
Few minor changes, and I don't know if we typically capitalize "Dashboard" or not, but LGTM

File: content/workers/configuration/compatibility-dates.md (hunk 3 above)
Comment (tanushree-sharma) [reply content/workers/configuration/compatibility-dates.md]: Some listed here https://developers.cloudflare.com/cloudflare-for-platforms/workers-for-platforms/reference/metadata/ though it isn't a complete list. On my to-do list to make it cover all metadata options. 

Our API docs don't have a good way of adding this file directly on there so going to (1) move https://developers.cloudflare.com/cloudflare-for-platforms/workers-for-platforms/reference/metadata/ under Workers and link to from WFP docs and (2) add a link to the dev docs for metadata.json from API docs. 

Merging this and will handle followups in a separate PR.

**Pull request merged**